datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -255,39 +256,61 @@ where
             if start > to {
                 return None;
             }
-            let end = start + step;
-            debug!(logger, "Requesting logs for blocks [{}, {}]", start, end);
+
+            // This will iterate more than once iff `step` has been reduced,
+            // making as many parallel requests of size `step` as necessary,
+            // respecting `LOG_STREAM_PARALLEL_CHUNKS`.
+            let mut chunk_futures = vec![];
+            let mut offset = start;
```

Comment:
`lower` and `upper` would be clearer here instead of `offset` and `next_offset` since they aren't really offsets from `start`

Reply:
true, this isn't an offset at all.